### PR TITLE
Mcp server setup

### PR DESCRIPTION
**Pull request by baz-propel**
Create structured codebase to allow for candidates to do challenges

### CODE REVIEW

File: docs/CHALLENGE_JUNIOR_MENU_MANAGEMENT.md [hunk 1]
```diff
@@ -0,0 +1,336 @@
+# Junior Challenge: Secret Menu System 🔒
+
+## Challenge Overview
+**Duration**: 2-3 hours  
+**Level**: Junior Developer  
+**Focus**: MCP Tool Development with Time-Based Features in C#
+
+Welcome! In this challenge, you'll build a Secret Menu System using the Model Context Protocol (MCP). This tests your understanding of timezone handling, time-based access control, and creative problem-solving in C#.
```
Comment (abdidaudpropel) [inline docs/CHALLENGE_JUNIOR_MENU_MANAGEMENT.md:8]: Suggest: to add understanding of MCP itself in what being tests (since it is an AI role)

File: docs/CHALLENGE_JUNIOR_MENU_MANAGEMENT.md [hunk 2]
```diff
@@ -0,0 +1,336 @@
...
+
+Think about trade-offs between simplicity and functionality!
+
+### 3. Time Handling Requirements
+- **Must use Melbourne timezone** (Australia/Melbourne)
+- Handle daylight saving transitions correctly
+- Display times in user-friendly format
+- Calculate "time until available" accurately
+- Use `TimeZoneInfo` or `NodaTime` for timezone conversions
+
```
Comment (abdidaudpropel) [inline docs/CHALLENGE_JUNIOR_MENU_MANAGEMENT.md:117]: leave out as well; finding out best practices on their own

File: docs/CHALLENGE_JUNIOR_MENU_MANAGEMENT.md [hunk 3]
```diff
@@ -0,0 +1,336 @@
...
+- Mystery ingredients
+- Item themes (can be restaurant-specific)
+- Limited quantity mechanics
+- Special tags or categories
+
+### 5. Validation Requirements
+- Restaurant ID must exist
+- Time validation must be precise to the minute
+- Handle timezone conversion errors gracefully
+- Validate that secret items don't conflict with regular menu
```
Comment (abdidaudpropel) [inline docs/CHALLENGE_JUNIOR_MENU_MANAGEMENT.md:131]: Leave out: part of what is measure is if candidate think of validation; another internal score

File: docs/CHALLENGE_JUNIOR_MENU_MANAGEMENT.md [hunk 4]
```diff
@@ -0,0 +1,336 @@
...
+        // Test with non-existent restaurant ID
+    }
+
+    [TestMethod]
+    public void GetSecretMenu_DaylightSavingTransition_HandlesCorrectly()
+    {
+        // Test during DST transitions
+    }
+}
+```
```
Comment (abdidaudpropel) [inline docs/CHALLENGE_JUNIOR_MENU_MANAGEMENT.md:171]: leave out

File: docs/CHALLENGE_JUNIOR_MENU_MANAGEMENT.md [hunk 5]
```diff
@@ -0,0 +1,336 @@
...
+```csharp
+public class SecretMenuItem : MenuItem
+{
+    public string MysteryIngredient { get; set; } = "???";
+    public int AvailableQuantity { get; set; }
+    public DateTime LastAvailable { get; set; }
+    public string SpecialStory { get; set; } // Fun backstory about the dish
+}
+```
+
```
Comment (abdidaudpropel) [inline docs/CHALLENGE_JUNIOR_MENU_MANAGEMENT.md:222]: leave out

File: docs/CHALLENGE_JUNIOR_MENU_MANAGEMENT.md [hunk 6]
```diff
@@ -0,0 +1,336 @@
...
+// Manual test example
+var result = await secretMenuTool.ExecuteTool("secret_menu_get", new
+{
+    restaurantId = "rest_001"
+});
+
+// Should return secret menu if between 3-4 PM Melbourne time
+// Should return friendly error message otherwise
+```
+
```
Comment (abdidaudpropel) [inline docs/CHALLENGE_JUNIOR_MENU_MANAGEMENT.md:285]: Leave out

File: docs/CHALLENGE_JUNIOR_MENU_MANAGEMENT.md [hunk 7]
```diff
@@ -0,0 +1,336 @@
...
+Before submitting, ensure:
+- [ ] `secret_menu_get` tool works correctly
+- [ ] Melbourne time validation is accurate
+- [ ] At least 5 unit tests pass
+- [ ] Secret items are creative and engaging
+- [ ] Code handles edge cases gracefully
+- [ ] Time window (3-4 PM) is enforced
+- [ ] Code follows C# conventions
+- [ ] No hardcoded times (use proper timezone conversion)
+
```
Comment (abdidaudpropel) [inline docs/CHALLENGE_JUNIOR_MENU_MANAGEMENT.md:311]: leave out; internal

File: docs/CHALLENGE_JUNIOR_MENU_MANAGEMENT.md [hunk 8]
```diff
@@ -0,0 +1,336 @@
...
+3. How would you prevent abuse of the time-limited system?
+4. What analytics would be useful for restaurant owners?
+5. How would you scale this for global restaurants with local secret menu times?
+
+## Resources
+
+- [TimeZoneInfo Documentation](https://docs.microsoft.com/en-us/dotnet/api/system.timezoneinfo)
+- [NodaTime Library](https://nodatime.org/) (optional, for advanced timezone handling)
+- [MCP Documentation](https://modelcontextprotocol.io)
+- [C# Unit Testing with MSTest](https://docs.microsoft.com/en-us/dotnet/core/testing/unit-testing-with-mstest)
```
Comment (abdidaudpropel) [inline docs/CHALLENGE_JUNIOR_MENU_MANAGEMENT.md:326]: leave out; part of take home test is exploring the above by themselves; and same goes with the questions to consider

File: docs/CHALLENGE_JUNIOR_MENU_MANAGEMENT.md [hunk 9]
```diff
@@ -0,0 +1,336 @@
...
+
+## Need Help?
+
+If you get stuck:
+1. Start with hardcoded secret items, then add creativity
+2. Test timezone logic separately first
+3. Use debugging to verify Melbourne time calculations
+4. Remember: creative and fun is better than complex
```
Comment (abdidaudpropel) [inline docs/CHALLENGE_JUNIOR_MENU_MANAGEMENT.md:334]: Very helpful but want to see how they approach the problem themselves and what workflow they have
